pkgs/applications/networking/remote/x2goserver/default.nix

```diff
@@ -0,0 +1,79 @@
+{ stdenv, lib, fetchurl, makeWrapper, perl, perlPackages, which, nx-libs
+, utillinux, coreutils, glibc, gawk, gnused, gnugrep, findutils, xorg
+, nettools, iproute, bc, procps, psmisc, lsof, pwgen, openssh, sshfs
+}:
+
+let binaryDeps = [
+  perl which nx-libs utillinux coreutils glibc.bin gawk gnused gnugrep
+  findutils nettools iproute bc procps psmisc lsof pwgen openssh sshfs
+  xorg.xauth xorg.xinit xorg.xrandr xorg.xmodmap xorg.xwininfo xorg.fontutil
+  xorg.xkbcomp xorg.setxkbmap
+]; in
+stdenv.mkDerivation rec {
+  pname = "x2goserver";
+  version = "4.1.0.3";
+
+  src = fetchurl {
+    url = "http://code.x2go.org/releases/source/x2goserver/${pname}-${version}.tar.gz";
+    sha256 = "1l6wd708kbipib4ldprfiihqmj4895nifg0bkws4x97majislxk7";
+  };
+
+  buildInputs = binaryDeps;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  patches = [ ./xsession-nixos-support.patch ];
+
+  preConfigure = ''
+    patchShebangs .
+    perl Makefile.PL
+    for i in */Makefile; do
+      substituteInPlace "$i" --replace "-o root -g root " ""
+    done
+    substituteInPlace libx2go-server-db-perl/Makefile --replace "chmod 2755" "chmod 755"
+    for i in X2Go/Config.pm x2goserver/sbin/x2godbadmin x2goserver/bin/x2go*
+    do
+      substituteInPlace $i --replace '/etc/x2go' "/var/lib/x2go/conf"
+    done
+    substituteInPlace X2Go/Server/DB.pm \
+      --replace '$x2go_lib_path/libx2go-server-db-sqlite3-wrapper' \
+                '/run/wrappers/bin/x2gosqliteWrapper'
+    substituteInPlace x2goserver/sbin/x2godbadmin --replace 'user="x2gouser"' 'user="x2go"'
+    substituteInPlace X2Go/Server/DB/SQLite3.pm --replace "user='x2gouser'" "user='x2go'"
+  '';
+
+  makeFlags = [ "PREFIX=/" "NXLIBDIR=${nx-libs}/lib/nx" "PERL=${perl}/bin/perl" "PERL_INSTALLDIRS=vendor" ];
+
+  installFlags = [ "DESTDIR=$(out)" ];
+
+  perlDeps = with perlPackages; [ DBI DBDSQLite FileBaseDir TryTiny CaptureTiny ConfigSimple Switch ];
+
+  perlInc = lib.concatMapStringsSep " " (dep: "-I ${dep}/${perlPackages.perl.libPrefix}") perlDeps;
+
+  postInstall = ''
+    mv $out/etc/x2go/x2goserver.conf{,.example}
+    mv $out/etc/x2go/x2goagent.options{,.example}
+    ln -sf ${nx-libs}/bin/nxagent $out/bin/x2goagent
+    for i in $out/sbin/x2go* $(find $out/bin -type f) \
+      $(ls $out/lib/x2go/x2go* | grep -v x2gocheckport)
+    do
+      wrapProgram $i \
+        --prefix PATH : ${lib.makeBinPath binaryDeps} \
+        --set PERL5LIB "${perlPackages.makeFullPerlPath perlDeps}:$out"
+    done
+    # We're patching @INC of the setgid wrapper, because we can't mix
+    # wrapProgram (for PERL5LIB) with security.wrappers (for setgid)
+    sed -i "s,bin/perl -XU,bin/perl $perlInc -I $out -XU," \
```

Comment:
Might be worth using `perl.withPackages` here like so:
```
perlEnv = perl.withPackages (p: with p; [ DBI DBDSQLite FileBaseDir TryTiny CaptureTiny ConfigSimple Switch ]);
```
Then you will have a `perl` binary (`perlEnv.interpreter`) which has everything you need, without the need to mess with `-I` and `PERL5LIB`.

Reply:
I did, unfortunately the perl-wrapper also works via an extra PERL5LIB shell wrapper, and because of that indirection the setguid x2go is lost again. But at least I can now patch in the environment site_perl, which is a lot shorter.